ddl2cpp: Cleanup part 2

MeanSquaredError opened the pull request:
This is the second part of the ddl2cpp cleanup. Changes:

- Most of the global functions and variables and grouped into three big classes: DdlParser, SelfTest and ModelWriter. Ideally these three would be modules instead of classes, but using modules would mean that the program code would be split into several files and we really want to avoid that because it would complicate the program installation and updates. So we use classes as poor man's version of modules.
- When parsing of multiple DDL files was fixed (https://github.com/rbock/sqlpp23/pull/62), another bug was introduced that broke the creation of split headers. So this PR also fixes this newly introduced bug. A note to myself: We should probably add ddl2cpp tests for the generation of headers, split headers and modules. But I guess this will have to wait until someone finds the time to add these tests.
- A for-loop was replaced with a list comprehension.

rbock commented on scripts/sqlpp23-ddl2cpp:
```diff
@@ -46,6 +46,12 @@ def static_init(cls):
     return cls
 
 
+# Ideally DdlParser, SelfTest and ModelWriter would be modules instead of classes. However using
```
As you explain in this comment, modules would not be "ideal" for our use case. And thus, we also don't need a "poor man's version". We use the approach that fits best.

```
# DdlParser, SelfTest and ModelWriter could be modules instead of classes. However, using
# modules implies that the program code would be split into several files and we want to avoid that
# because it would complicate program installation. That is why we use classes.
```

MeanSquaredError replied:
OK, I updated the comment. No more ideal poor man's versions.

MeanSquaredError commented:
I updated the PR:
- Changed the comment about the class `DdlParser`, `SelfTest` and `ModelWriter`.
- Fixed a bug that caused incorrect SQL commands to be emitted when generating the table creation helper. The reason for the bug was that the parse action (in our case `AddCreateSql`) was receiving the DDL text for the whole file, not just the part matching the CREATE command. I guess there is no way to avoid using `pyparsing.Located`...
- Disabled expansion of tabs into spaces. By default pyparsing expands tabs into spaces in the input text. In our case this was clearly unwanted behavior because it could cause the generation of incorrect table creation helpers. For example the following table definition `CREATE TABLE tbl (mytext text NOT NULL DEFAULT '\t\t')` would have the tabs expanded during parsing and the wrong table creation code would be emitted (having spaces instead of tabs).

I tried to do some additional manual testing of the script (I generated single header, split headers, modules and checked the generated C++ code manually). The generated headers/modules seem to be correct, but I think that we should add automatic tests for this in the future.

MeanSquaredError commented:
It looks like the PR broke the support for extended types. Marking it as Draft until I fix it.

MeanSquaredError commented:
It will take me some time to provide a proper fix for the loading of custom data types. Since I will be busy in the next 7-10 days, this PR will be on hold for the time being. So I decided to provide just the bugfixes as a separate PR in https://github.com/rbock/sqlpp23/pull/73.

MeanSquaredError commented:
Closing this PR as it's been superseded by https://github.com/rbock/sqlpp23/pull/74 and https://github.com/rbock/sqlpp23/pull/76
